### ISSUE TITLE
Possibly 3830 sector format?

### ISSUE THREAD

**Issue by pkooiman** [schlae/minnow-disk#1]
Hi Eric, what a cool project! I was messing around with the track 0 data, and I could not make it match with the 370/135 sector description int the IBM doc. If I read it correctly, each byte should immediately be followed by the next start bit (so startbit, 8xdatabit, parity bit, startbit) but the track data seems to have lots of extra 0 bits between bits. Also, lots and lots of parity errors.

Now, treating it as if it were 3830 format, so 1 sync bit at sector start, then a continuous stream of 258 bytes consisting of  8 x databit, + 1 paritybit, it all seems to line up, and no parity errors on track 0.

Having said that, all the other tracks do give lots of parity errors (though some of them are mostly error free). I find it hard to believe that track0 would have no parity errors by pure coincidence! So I am wondering, is there any chance the disk is actually 3830 format?

Peter

**Comment by schlae** [comment]
Someone else pointed out that the documentation I've been using is for the model 145, which is not the same as the disk, which is meant for the 135. It seems highly likely that the 135 uses a different format, and perhaps a simpler disk controller design. I wouldn't be surprised if it was similar to the 3830. Unfortunately there's not much we can do with this unless documentation for the 135 shows up.

**Comment by pkooiman** [comment]
Right, that would explain it. It's funny, the whole of track 0 and the first 3 sectors of track one decode perfectly with 3830 format, but for the rest of the disk it gives the 50% parity error rate you would expect from pure chance. It's almost as if the encoding scheme changes after track1/sector2. The dump itself seems fine with tight interval groupings around 4 and 8 usec on all tracks. Ah well, who knows, 135 documentation may show up someday!

**Comment by rzsn** [comment]
I applied the same method for every sector - and except the first 11 of parity-ok sectors for every byte:

```
00+ 4d+ 70+ 06+ 00+ 00+ 4d+ 70+ 46+ 11+ 11+ 4d+ ...
01+ 0d+ 49+ 9a+ 7d+ 80+ 4d+ 71+ 86+ ca+ 53+ 1d+ ...
02+ 0d+ 49+ 9b+ 7d+ 80+ 0d+ 55+ f1+ 4b+ 36+ 0d+ ...
03+ 0d+ 49+ 9c+ 7d+ 80+ 0d+ 4f+ 58+ 5f+ 1f+ 0d+ ...
04+ 0d+ 49+ 9d+ 7d+ 80+ 0d+ 66+ eb+ 63+ 5a+ 0d+ ...
05+ 0d+ 49+ 9e+ 7d+ 80+ 0d+ 4e+ d8+ 67+ 59+ 0d+ ...
06+ 0d+ 49+ 9f+ 7d+ 80+ 0d+ 4e+ 98+ 4e+ 58+ 0d+ ...
07+ 0d+ 58+ 58+ 7c+ 40+ 0d+ 42+ 5a+ 40+ 4e+ 0d+ ...
08+ 0d+ 58+ 59+ 7c+ 40+ 0d+ 6c+ 59+ 50+ 68+ 0d+ ...
09+ 4d+ 70+ 46+ 00+ 00+ 4d+ 30+ 4b+ 4b+ 30+ cd+ ...
0a+ 4d+ 70+ 46+ 30+ 80+ 4d+ 30+ 4b+ ce+ 8c+ cd+ ...
```

... starting from the 12-th sector there is a pattern on beginning of each of them:
```
0b+ 98+ 00+ 00+ 00+ 09+ 0c+ fb+
0c+ 98+ 00+ 01+ 30+ 09+ 0d+ 1a+
0d+ 98+ 00+ 02+ 60+ 09+ 0e+ 37+
0e+ 98+ 00+ 03+ 90+ 09+ 0f+ ec+
0f+ 98+ 00+ 04+ c0+ 09+ 10+ 8b+
10+ 98+ 00+ 05+ f0+ 09+ 11+ 1c+
11+ 98+ 00+ 07+ 20+ 09+ 12+ 8a+
12+ 98+ 00+ 08+ 50+ 09+ 13+ 0f+
13+ 98+ 00+ 09+ 80+ 09+ 14+ a9+
:
[1] [2] [3] [4] [5] [6] [7] [8]
:
f6+ 98+ 00+ d5+ 80+ 09+ f7+ e3+
f7+ 98+ 00+ d6+ b0+ 09+ f8+ 40+
f8+ 98+ 00+ d7+ e0+ 09+ f9+ 07+
f9+ 98+ 00+ d9+ 10+ 09+ fa+ 95+
fa+ 98+ 00+ da+ 40+ 09+ fb+ 06+
fb+ 98+ 00+ db+ 70+ 09+ fc+ f2+
fc+ 98+ 00+ dc+ a0+ 09+ fd+ 4d+
fd+ 98+ 00+ dd+ d0+ 09+ fe+ af+
fe+ 81+ 00+ df+ 00+ 01+ ff+ b0+
ff+ 00+ 00+ 00+ 00+ 00+ 00+ 00+
```

There are **always at least 8 of these starting bytes** = _the sector header_, and then the sector does not always have more data bytes (they are not zeroes, since the parity bit does not appear past the 8 header bytes)

Bytes:

1. **LBA address (5+3 bits of track+sector)**
2. looks like an opcode/flags (0x00,0x01,0x38,0x80,0x81,0x98)
3. **always 0x00**
4. some destination address? (0x00 - 0xDF with missing codes)
5. some address extension? (0x00,0x10,0x20...0xF0 - all 16 high nibble codes)
6. some flags? (0x00,0x01,0x09)
7. **LBA of next sector** !
8. some sort of CRC ?
